### ISSUE TITLE
[a11y]: Export Modal, unable to close dialog using esc key, after exporting.

### ISSUE THREAD

**Issue by devadula-nandan** [carbon-design-system/ibm-products#6016]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.49.0-rc.0

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during the QA Audit

After exporting the file, a success message appears in same dialog. Unable to close dialog using esc.
The Modal expected to close with a press of Escape key 

### WCAG 2.1 Violation

2.1.1. Keyboard access 

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-export-exportmodal--standard&globals=viewport:basic

### Steps to reproduce

This was reported in all variants of the Export Modal.
open export modal, and click on export button, wait for success message. try to close the modal with esc key and notice it doesn't close

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by devadula-nandan** [comment]
it looks like the focus is getting reset to the default body due to a re-render.
<img width="1477" alt="Screenshot 2024-09-06 at 2 21 51 PM" src="https://github.com/user-attachments/assets/840d4c32-a1af-4b85-8ee5-d35e4422bf03">

**Comment by anamikaanu96** [comment]
Focus issue is solved via [issue](https://github.com/carbon-design-system/ibm-products/issues/6021), now the Dialog is closing after pressing esc 2 times. This is the expected behaviour since export modal is using composed modal from carbon.

Behaviour of composed modal in carbon.

https://github.com/user-attachments/assets/39f419ff-7c38-46b4-a34b-e5aed2639f17

**Comment by amal-k-joy** [comment]
Hi @devadula-nandan ,
As confirmed by the carbon team on this [slack discussion](https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1727762965870589), its the expected behaviour to close modal on second escape only 

_

> After discussing with the team, we clarified the hierarchy of the ESC functionality, which typically ranges from the most specific to broader applications. For instance, a user might prefer to dismiss the tooltip only while still being able to read the modal.
> In short, this is the expected behaviour.

_
